x-pack/plugins/rule_registry/server/rule_data_client/rule_data_client.ts

```diff
@@ -181,43 +181,43 @@ export class RuleDataClient implements IRuleDataClient {
       }
     };
 
-    const prepareForWritingResult = prepareForWriting();
+    const prepareForWritingResult = prepareForWriting().catch((error) => {
+      if (error instanceof RuleDataWriterInitializationError) {
+        this.options.logger.error(error);
+        this.options.logger.error(
+          `The writer for the Rule Data Client for the ${indexInfo.indexOptions.registrationContext} registration context was not initialized properly, bulk() cannot continue, and writing will be disabled.`
+        );
+        turnOffWrite();
+      } else if (error instanceof RuleDataWriteDisabledError) {
+        this.options.logger.debug(`Writing is disabled, bulk() will not write any data.`);
+      }
+      return undefined;
+    });
 
     return {
       bulk: async (request: estypes.BulkRequest) => {
-        return prepareForWritingResult
-          .then((clusterClient) => {
+        try {
+          const clusterClient = await prepareForWritingResult;
+          if (clusterClient) {
```

Comment:
Previously a debug message `Writing is disabled, bulk() will not write any data.` was logged each time `bulk()` is called if writing is disabled, but now the message is only logged when the writer is initialized for the first time. Should we continue to log a message every time `bulk()` is called if `clusterClient` is undefined?